src/pipx/venv.py

```diff
@@ -97,12 +97,13 @@ def __init__(self, path: Path, *, verbose: bool = False, python: str = DEFAULT_P
         except StopIteration:
             self._existing = False
 
+    def check_upgrade_shared_libs(self, pip_args: List[str], verbose: bool):
```

Comment:
```suggestion
    def check_upgrade_shared_libs(self, pip_args: Optional[List[str]] = None, verbose: bool):
```

Reply:
I don't know the codebase well, so I'm confused by your suggestion here. (I may be missing contribution guidelines that clarifiy this - and I definitely lack experience in maintenance of widely-used open-source projects. )

If I understand, we agree on the fact that the point of the MR is to get rid of scenarios where shared libs operations ignore user-provided pip args under the hood.

So, isn't it "dangerous" to accept `None` as a default value here ? I felt that *not* providing an acceptable value would better document to future developpers that calling this function is not acceptable if we don't have  a user-provided `pip_args`

Reply:
If we stick with `pip_args: List[str]` I think there's a lot of implicit meaning contained in that parameter. We should add a docstring that explains the logic behind it (commands that allow modifying the `pip_args` providing it as an empty list, even if there's no arguments provided)

Reply:
Probably it _is_ better to be on the safe side and move the shared libs upgrade into a dedicated command, as said in https://github.com/pypa/pipx/issues/609#issuecomment-765833354. While that would be mutually exclusive with a global `--skip-maintenance` option. Introducing the latter, we could document to pass it whenever one does not want to forward the `pip` args to the `pipx` shared libs upgrade, which I think is a practical solution, since  
upgrading the lib by hand would mean extra efforts for the users and could be easily forgotten.

Generally, I doubt users will often make use of complex setups with `pipx` that could damage the shared library in any way. So I think this solution is good for now.